### PR TITLE
1787: Refactor new message component

### LINKED ISSUE #1787: Refactor NewMessage to be component based
We currently only use `<NewMessage/>` from the `client` package, where it is stored.

But, I need to access it from the `components` package. So, we should refactor it to match the structure of our `components`, and create a story/spec to test it.

If possible, we should refactor `MessageCreator` to the new `Component` model at the same time.

It looks like both elements are already TypeScript, we just need to match our new way of building things.

We could (temporarily) call the new version `<NewMessage2/>` during testing. Once it's accepted, we drop the existing `NewMessage` and rename `NewMessage2` to `NewMessage`.

### PR DESCRIPTION
**Pull request by lebaphi**
Fixes #1787

### CODE REVIEW

**Comment by IanMayo** [comment]
Thanks for this @lebaphi 

The Heroku build is failing. The same error prevents me running `yarn storybook`.  My guess is that it's failing with this import. Experimenting.
image

**Comment by lebaphi** [comment]
yep, I will check this 🤔

**Comment by IanMayo** [comment]
Hello @lebaphi - I've fixed the build issue.  But, before I could `push` I had loads of linter fixes to do.  I _think_ I've got them right.  I've tried to produce a `storybook` for `new-message`, but it's not quite working yet.

In `yarn develop`, it looks like I can't send a new message in `Papa Foxtrot` when I'm logged in as `White - Peoples` in the `White Chat` channel.

I _guess_ it needs more work, but we're getting there :-)

**Comment by lebaphi** [comment]
> In yarn develop, it looks like I can't send a new message in Papa Foxtrot when I'm logged in as White - Peoples in the White Chat channel.

ahh sorry, I forgot to remove the debug flag, I think it works now

>I've tried to produce a storybook for new-message, but it's not quite working yet.

I'm working on it now

**Comment by lebaphi** [comment]
I think component works but the style is not applied, because NewMessage2 is using global style the same way it works in client packages, so we need to re-implement the style for this component, as component scope, what do you think? @IanMayo 


https://user-images.githubusercontent.com/26705621/189467298-6f076d63-1500-489f-82da-2f7623c4813b.mp4

**Comment by IanMayo** [comment]
>  so we need to re-implement the style for this component, as component scope, what do you think?

Sorry @lebaphi - I don't know the answer. What do you recommend?

**Comment by lebaphi** [comment]
@IanMayo, I mean NewMessage2 doesn't use styles imported from styles.module.scss like other components, it uses styles from common css files. Therefore, it only works with yarn develop, not storybook. So if you want to apply styles in storybook, we need to migrate the style from global to style.module.scss (component scope).

**Comment by IanMayo** [comment]
it would be good for stories to have proper styling. I'll review this again tomorrow.
You can make the styles changes in this branch, if you like :-)

File: packages/components/src/local/form-elements/new-message/index.stories-inc.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    confirmCancel: {
+      description: 'Whether player has to confirm cancelling a message',
+      control: {
+        type: 'boolean'
+      }
+    }
+  }
+}
+
+export const saveMessage = (dbName: string, details: MessageDetails, message: object): Function => (): Function => {
```
Comment (IanMayo) [inline packages/components/src/local/form-elements/new-message/index.stories-inc.tsx]: Aah, @lebaphi - I could do with help from you here, please.  I can't handle the T/S to pass `saveMessage` in the props below. Could you help me out, and get it working please?

Comment (lebaphi) [reply packages/components/src/local/form-elements/new-message/index.stories-inc.tsx]: sure @IanMayo

**Comment by lebaphi** [comment]
> Aah, @lebaphi - I could do with help from you here, please. I can't handle the T/S to pass saveMessage in the props below. Could you help me out, and get it working please?

updated @IanMayo

**Review by IanMayo — APPROVED** [review]

**Pull request merged**